hscript/Interp.hx

```diff
@@ -526,18 +597,48 @@ class Interp {
 		restore(old);
 	}
 
+	inline function isMap(o:Dynamic):Bool {
+		return Std.is(o, haxe.Constraints.IMap);
+	}
+	
+	inline function getMapValue(map:Dynamic, key:Dynamic):Dynamic {
+		return cast(map, haxe.Constraints.IMap<Dynamic, Dynamic>).get(key);
+	}
+
+	inline function setMapValue(map:Dynamic, key:Dynamic, value:Dynamic):Void {
+		cast(map, haxe.Constraints.IMap<Dynamic, Dynamic>).set(key, value);
+	}
+	
 	function get( o : Dynamic, f : String ) : Dynamic {
-		if( o == null ) error(EInvalidAccess(f));
-		#if php
-			// https://github.com/HaxeFoundation/haxe/issues/4915
-			return try {
+		if ( o == null ) error(EInvalidAccess(f));
+		var result:Dynamic = {
+			#if php
+				// https://github.com/HaxeFoundation/haxe/issues/4915
+				try {
+					Reflect.getProperty(o, f);
+				} catch (e:Dynamic) {
+					Reflect.field(o, f);
+				}
+			#else
 				Reflect.getProperty(o, f);
-			} catch (e:Dynamic) {
-				Reflect.field(o, f);
+			#end
+		}
+		
+		if (result == null && isMap(o)) {
+			var map = cast(o, haxe.Constraints.IMap<Dynamic, Dynamic>);
+			result = switch(f) {
+				case 'get': map.get;
+				case 'set': map.set;
+				case 'exists': map.exists;
+				case 'remove': map.remove;
+				case 'keys': map.keys;
+				case 'iterator': map.iterator;
+				case 'toString': map.toString;
+				default: null;
 			}
-		#else
-			return Reflect.getProperty(o, f);
-		#end
+		}
+		

```

Comment:
That shouldn't be the case. Reflect.hasField will only return true for anonymous structures, it will not work on some platforms if you access a method. Reflect.field however will return the correct value. Maybe that's why you have a misunderstanding?